_posts/2021-10-25-waku-v1-vs-waku-v2.md

```diff
@@ -0,0 +1,202 @@
+discuss: https://forum.vac.dev/t/discussion-waku-v1-vs-waku-v2-bandwidth-comparison/110
+---
+
+## Background
+
+The [original plan](https://vac.dev/waku-v2-plan) for Waku v2 suggested theoretical improvements in resource usage over Waku v1,
+mainly as a result of the improved amplification factors provided by GossipSub.
+In its turn, [Waku v1 proposed improvements](https://vac.dev/fixing-whisper-with-waku) over its predecessor, Whisper.
+
+Given that Waku v2 is aimed at resource restricted environments,
+we are specifically interested in its scalability and resource usage characteristics.
+However, the theoretical performance improvements of Waku v2 over Waku v1,
+has never been properly benchmarked and tested.
+
+A full performance evaluation of Waku v2 would require significant planning and resources,
+if it were to simulate "real world" conditions faithfully and measure bandwidth and resource usage across different network connections,
+robustness against attacks/losses, message latencies, etc.
+(There already exists a fairly comprehensive [evaluation of GossipSub v1.1](https://research.protocol.ai/publications/gossipsub-v1.1-evaluation-report/vyzovitis2020.pdf),
+on which `11/WAKU2-RELAY` is based.)
+
+As a more realistic starting point,
+this document contains a limited and local comparison of the _bandwidth_ profile (only) between Waku v1 and Waku v2.
+It reuses and adapts existing network simulations for [Waku v1](https://github.com/status-im/nim-waku/blob/master/waku/v1/node/quicksim.nim) and [Waku v2](https://github.com/status-im/nim-waku/blob/master/waku/v2/node/quicksim2.nim)
+and compares bandwidth usage for similar message propagation scenarios.
+
+## Theoretical improvements in Waku v2
+
+Messages are propagated in Waku v1 using [flood routing](https://en.wikipedia.org/wiki/Flooding_(computer_networking)).
+This means that every peer will forward every new incoming message to all its connected peers (except the one it received the message from).
+This necessarily leads to unnecessary duplication (termed _amplification factor_),
+wasting bandwidth and resources.
+What's more, we expect this effect to worsen the larger the network becomes,
+as each _connection_ will receive a copy of each message,
+rather than a single copy per peer.
+
+Message routing in Waku v2 follows the `libp2p` _GossipSub_ protocol,
+which lowers amplification factors by only sending full message contents to a subset of connected peers.
+As a Waku v2 network grows, each peer will limit its number of full-message ("mesh") peerings -
+`libp2p` suggests a maximum of `12` such connections per peer.
+This allows much better scalability than a flood-routed network.
+From time to time, a Waku v2 peer will send metadata about the messages it has seen to other peers ("gossip" peers).
+
+See [this explainer](https://hackmd.io/@vac/main/%2FYYlZYBCURFyO_ZG1EiteWg#11WAKU2-RELAY-gossipsub) for a more detailed discussion.
```

Comment:
Gah, this url is not very pretty, seems like it can become the victim of bit rot. I don't see a better alternative right now though...